### ISSUE TITLE
Allowing / in patch names

### ISSUE THREAD

**Issue by lthms** [stacked-git/stgit#402]
As part of my professional workflow, I’d like to have a naming convention for my patch that’d be `<handle>@<project>/<task>`, so for instance `lthms@shell/fix-bug`.

Today, Stacked Git does not allow `/` in patch name: is it an intrinsic limitation (which cannot be overcome due to how Stacked Git works internally) or is it a limit we could relax?

**Comment by jpgrayson** [comment]
Here is something I wrote on this topic a while back: https://github.com/stacked-git/stgit/issues/24#issuecomment-923311009

The TL;DR is that allowing slashes in patch names _can_ be achieved, but there are costs of complexity and risk.

One area where I know there will be challenges is with maintaining the per-patch refs found in `refs/patches`. The per-patch refs are arranged with this schema: `refs/patches/<branchname>/<patchname>`. With the current naming rules, a branch name may have slashes, but since the patch name cannot have slashes, we always know that the basename (leaf) of these refs is the patch name.

If patch names could have slashes, then we become exposed to parsing ambiguities and potential name collisions. I think these are all solvable problems, but it will take some care to cover all the edge cases correctly.

And I expect there will be other places where current assumptions that patch names don't have slashes will cause trouble. Really anywhere we might want to map a patch name to a filename will be trouble because a patch would now more generally map to a path fragment instead of a plain filename. This probably affects import and export; maybe other commands.

**Comment by lthms** [comment]
That makes sense, thanks for explaining! I’ll replace `/` by `--` and call it a day :sweat_smile:

**Comment by bonki** [comment]
@jpgrayson I understand that this is non-trivial but would you consider re-opening this?
